tcti/tcti_socket.c

```diff
@@ -528,3 +534,65 @@ TSS2_RC InitSocketTcti (
 
     return rval;
 }
+
+TSS2_RC Tss2_Tcti_Socket_Init (
+    TSS2_TCTI_CONTEXT *tctiContext,
+    size_t *size,
+    const char *conf
+    )
+{
+    TCTI_SOCKET_CONF sock_conf = { 0 };
+    TSS2_RC rc;
+    UriParserStateA state;
+    UriUriA uri;
+    const char *uri_str = conf != NULL ? conf : TCTI_SOCKET_DEFAULT;
+    size_t range;
+    /* maximum 5 digits in uint16_t + 1 for \0 */
+    char port[6] = { 0 };
+    char hostname[HOST_NAME_MAX + 1] = { 0 };
+
+    state.uri = &uri;
+    if (uriParseUriA (&state, uri_str) != URI_SUCCESS) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+
+    /* extract host & domain name / fqdn */
+    range = uri.hostText.afterLast - uri.hostText.first;
+    if (range > HOST_NAME_MAX) {
+        rc = TSS2_TCTI_RC_BAD_VALUE;
+        goto out;
+    }
+    strncpy (hostname, uri.hostText.first, range);
+    sock_conf.hostname = hostname;
+
+    /* extract port number */
+    range = uri.portText.afterLast - uri.portText.first;
+    if (range < 5 && range > 0) {
+        strncpy (port, uri.portText.first, range);
```

Comment:
Need to verify if port is not greater than 65535? 99999 will not fit into uint16_t.

Reply:
Good catch. I'm putting it in a `uint32_t` now first and checking for `> UINT16_MAX` before putting it into a `uint16_t` for the call to `connect. PR updated.